pennsignals/gentry
# Update encode.go and encode_test.go

- Remove Product method from Converter interface
- Update BufferedReader.Parse method
- Update PostMessageEncoder.Encode method


**Verification steps:**

Test using native go binary:

    $ go test -v ./...

Test using a Docker container:

    $ docker run --rm -v "$PWD":/usr/src/gentry -w /usr/src/gentry golang go test -v ./...

See #29
Closes #31
